src/lake/Lake/Load/Lean/Elab.lean

```diff
@@ -259,6 +268,8 @@ def importConfigFile (cfg : LoadConfig) : LogIO Environment := do
     validateTrace <| ← IO.FS.Handle.mk traceFile .read
   else
     IO.FS.createDirAll cfg.lakeDir
+    -- Create CACHEDIR.TAG file only when .lake directory is first created
+    IO.FS.writeFile (cfg.lakeDir / "CACHEDIR.TAG") cachedirTagContent
```

Comment:
This is not really what this this does. If a user deletes the `CACHEDIR.TAG` (because they wish the directory to be cached) and then, unrelatedly, delete the configuration trace (e.g., due to an incomplete configuration), this will regenerate the `CACHEDIR.TAG` (which would be quite surprising behavior).

Now that creating the lake directory is something worth doing independently, I suggest refactoring out the `cfg.lakeDir` creation here to the top of this function (outside all the locking). That is, just add something like this to the top and remove it from here:

```lean
if !(<- cfg.lakeDir.pathExists) then
  IO.FS.createDirAll cfg.lakeDir
  -- Create CACHEDIR.TAG file only when .lake directory is first created
  IO.FS.writeFile (cfg.lakeDir / "CACHEDIR.TAG") cachedirTagContent
```